### PR TITLE
Fix/con 111/perfomance improvement

### PR DESCRIPTION
**Pull request by boajer**
Cherry pick of https://github.com/oat-sa/extension-tao-mediamanager/pull/239

https://oat-sa.atlassian.net/browse/CON-111

### CODE REVIEW

File: model/MediaSource.php [hunk 1]
```diff
@@ -118,16 +118,16 @@ public function getDirectory($parentLink = '', $acceptableMime = [], $depth = 1)
...
             foreach ($class->searchInstances($filter) as $instance) {
-                try {
-                    $file = $this->getFileInfo($instance->getUri());
-                    if (count($acceptableMime) == 0 || in_array($file['mime'], $acceptableMime)) {
-                        $children[] = $file;
-                    }
-                } catch (\tao_models_classes_FileNotFoundException $e) {
+                try{
+                    $children[] = $this->getFileInfo($instance->getUri());
+                }catch(\tao_models_classes_FileNotFoundException $e){
```
Comment (gabrielfs7) [inline model/MediaSource.php:130]: psr2 spaces missing

Comment (gabrielfs7) [reply model/MediaSource.php:130]: FQN

**Review by gabrielfs7 — APPROVED** [review]
Code is similar as previously tested fix here: https://github.com/oat-sa/extension-tao-mediamanager/pull/239

Now we have the integration test to be fixed, no? `testGetFileInfo()`

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/MediaSource.php [hunk 2]
```diff
@@ -118,16 +118,16 @@ public function getDirectory($parentLink = '', $acceptableMime = [], $depth = 1)
                 $children[] = $this->getDirectory($subclass->getUri(), $acceptableMime, $depth - 1);
             }
 
-            // add a filter for example on language (not for now)
             $filter = [];
 
+            if (count($acceptableMime) > 0) {
```
Comment (shpran) [inline model/MediaSource.php:123]:
```suggestion
            if (!empty($acceptableMime)) {
```

**Review by shpran — APPROVED** [review]
Well done 👍🏻 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/MediaSource.php [hunk 3]
```diff
@@ -147,46 +148,52 @@ public function getFileInfo($link)
     {
         // get the media link from the resource
         $resource = $this->getResource(tao_helpers_Uri::decode($this->removeSchemaFromUriOrLink($link)));
-        if (!$resource->exists()) {
-            throw new \tao_models_classes_FileNotFoundException($link);
-        }
+        $properties = [
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/MediaSource.php:151]: are you sure you want to hard code those proprieties ?

File: model/MediaSource.php [hunk 4]
```diff
@@ -147,46 +148,52 @@ public function getFileInfo($link)
...
+            $this->getProperty(MediaService::PROPERTY_LINK),
+            $this->getProperty(MediaService::PROPERTY_MIME_TYPE),
+            $this->getProperty(MediaService::PROPERTY_ALT_TEXT)
+        ];
 
-        $fileLink = $resource->getUniquePropertyValue($this->getProperty(MediaService::PROPERTY_LINK));
+        $propertiesValues = $resource->getPropertiesValues($properties);
+
+        $fileLink = $propertiesValues[MediaService::PROPERTY_LINK][0] ?? null;
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/MediaSource.php:159]: null or empty string so you don't have to cast ?

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: model/MediaSource.php (hunk 4 above)
Comment (gabrielfs7) [reply model/MediaSource.php:159]: It is casted on line 161

**Pull request merged**